docs/tls/CERTIFICATES.md

```diff
@@ -0,0 +1,152 @@
+# Self-signed certificates guide
+
+This guide describes the steps required to generate self-signed certificates for use with NGINX Plus and NLK.
+
+When using self-signed certificates, the first step is to generate the Certificate Authority (CA),
+which will be used to sign the certificates for the client and the server.
+The `openssl` utility will be used to generate the CA and the certificates.
+
+#### Generate the CA
+
+The following commands will generate the CA certificate and key:
+
+```bash
+openssl req -newkey rsa:2048 -nodes -x509 -out ca.crt -keyout ca.key
+```
+
+You will be prompted to enter the Distinguished Name (DN) information for the CA. Alternatively, you can provide the DN information in a file, an example is shown below:
+
+```bash
+[ req ]
+distinguished_name = dn
+prompt = no
+req_extensions = req_ext
+
+[ req_ext ]
+basicConstraints = CA:TRUE
+keyUsage = critical, keyCertSign, cRLSign
+
+[ dn ]
+C=[COUNTRY]
+ST=[STATE]
+L=[LOCALITY]
+O=[ORGANIZATION]
+OU=[ORGANIZATION_UNIT]
+```
+
+The `openssl` command can then be updated to use the DN information from the file:
+
+```bash
+openssl req -newkey rsa:2048 -nodes -x509 -config ca.cnf -out ca.crt -keyout ca.key
+```
+
+#### Generate the server certificate
+
+##### Generate the server certificate request
+
+The next step is to generate the server certificate and key. The certificate has the same basic field as the CA certificate,
+but with a few additional fields.
+
+The following is an example of a configuration file for server certificates:
+
+```bash
+[ req ]
+distinguished_name = dn
+req_extensions = req_ext
+prompt = no
+
+[ dn ]
+C=[COUNTRY]
+ST=[STATE]
+L=[LOCALITY]
+O=[ORGANIZATION]
+OU=[ORGANIZATION_UNIT]
+CN=[COMMON_NAME]
+
+[ req_ext ]
+basicConstraints = CA:FALSE
+keyUsage = nonRepudiation, digitalSignature, keyEncipherment
+subjectAltName = @alt_names
+extendedKeyUsage = serverAuth
+
+[ alt_names ]
+DNS.1 = mydomain.com
+DNS.2 = server.mydomain.com
+DNS.3 = *.mydomain.com
+IP.1 = 10.0.0.10
+IP.2 = 10.0.0.11
+```
+
+This example includes sensible defaults in terms of extensions, to learn more see the [OpenSSL extensions documentation](https://www.openssl.org/docs/manmaster/man5/x509v3_config.html).
+
+It is important to note that the Common Name (CN) field is deprecated for verification.
+The Subject Alternative Name (SAN) should be used instead. The SAN allows unambiguous bindings to IP addresses and/or domain names.
+
+Be sure to update the DN information and the SAN information (DNS / IP entries in the `alt_names` section) as appropriate.
```

Comment:
Does this require an explanation as to why this matters?
I think there is some domain knowledge wrapped in these statements that the reader might not hold.
Such as "defining an IP allow you to configure NLK using an IP instead of a DNS name" and so forth.

Reply:
Reworked this along with the above feedback.